extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java

```diff
@@ -612,6 +643,190 @@ BeanDefiningAnnotationBuildItem quarkusMain() {
+            List<InjectionPointInfo> injectionPoints, BuildProducer<ReflectiveMethodBuildItem> reflectiveMethods,
+            BuildProducer<ReflectiveFieldBuildItem> reflectiveFields,
+            BuildProducer<UnremovableBeanBuildItem> unremovableBeans) {
+        BeanDeployment beanDeployment = beanRegistrationPhase.getBeanProcessor().getBeanDeployment();
+        AnnotationLiteralProcessor annotationLiterals = beanRegistrationPhase.getBeanProcessor()
+                .getAnnotationLiteralProcessor();
+        ReflectionRegistration reflectionRegistration = new ReflectionRegistration() {
+            @Override
+            public void registerMethod(MethodInfo methodInfo) {
+                reflectiveMethods.produce(new ReflectiveMethodBuildItem(methodInfo));
+            }
+
+            @Override
+            public void registerField(FieldInfo fieldInfo) {
+                reflectiveFields.produce(new ReflectiveFieldBuildItem(fieldInfo));
+            }
+
+        };
+
+        Set<TypeAndQualifiers> unremovables = new HashSet<>();
+        for (InjectionPointInfo injectionPoint : injectionPoints) {
+
+            // The injection point must be registered immediately and NOT inside the creator callback
+            if (injectionPoint.isField()) {
+                reflectionRegistration.registerField(injectionPoint.getTarget().asField());
+            } else {
+                reflectionRegistration.registerMethod(injectionPoint.getTarget().asMethod());
+            }
+
+            // All qualifiers but @All and @Identified
+            Set<AnnotationInstance> qualifiers = injectionPoint.getRequiredQualifiers().stream()
+                    .filter(a -> !DotNames.ALL.equals(a.name()) && !DotNames.IDENTIFIED.equals(a.name()))
+                    .collect(Collectors.toSet());
+            if (qualifiers.isEmpty()) {
+                // If no other qualifier is used then add @Any
+                qualifiers.add(AnnotationInstance.create(DotNames.ANY, null, new AnnotationValue[] {}));
+            }
+
+            Type elementType = injectionPoint.getType().asParameterizedType().arguments().get(0);
+
+            if (!unremovables
+                    .add(new TypeAndQualifiers(
+                            elementType.name().equals(DotNames.INSTANCE_HANDLE)
+                                    ? elementType.asParameterizedType().arguments().get(0)
+                                    : elementType,
+                            qualifiers))) {
+                continue;
+            }
+
+            BeanConfigurator<?> configurator = beanRegistrationPhase.getContext()
+                    .configure(List.class)
+                    .forceApplicationClass()
+                    .scope(BuiltinScope.DEPENDENT.getInfo())
+                    .addType(injectionPoint.getRequiredType());
+
+            injectionPoint.getRequiredQualifiers().forEach(configurator::addQualifier);
```

Comment:
Just wondering if this should read the `qualifiers` set created above, instead of `injectionPoint.getRequiredQualifiers()`?

Reply:
I don't think so. The synthetic bean must copy the qualifiers on the injection point, e.g. `@All`, whereas the `qualifiers` created above are used for dynamic lookup when the list is created at runtime, e.g. `@Any`.

Reply:
Gotcha, thanks!